### PR TITLE
Config and Plugin Support

### PR DESCRIPTION
**Pull request by aemmadi**
This PR includes:

## Config

- Now scans `/:user/:repo` for a `.fastdocs.json` config file. This allows fastdocs to know which plugins to add when rendering the docs.
- Defaults to rendering without config if any error is thrown during parsing the config.

_Current Config_:
```json
{
  "description": "Description",
  "plugins": [],
  "gaCode": ""    
}
```
_`gaCode` is only required when `google-analytics` is added to the `plugins` array_

## Plugins
- Added support for most **major plugins** on docsify that were determined to be "lightweight" enough to be added to the fastdocs microservice. These include:
    - `docsify-copy-code`: Adds a "Copy to Clipboard" next to code blocks
    - `full-text-search`: Adds a search bar on the top of the sidebar to quickly search docs
    - `google-analytics`: Support for google analytics
       - Need to provide the google analytics code inside `gaCode` on the config.
    - `emoji`: Renders emoji's in your documentation. (Ex: `:100:` would render to 💯) 
    - `zoom-image`: Adds subtle zooming to images cursor is hovering

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -10,10 +11,16 @@ app.get("/:user/:repo/compile", (req, res) => {
...
+  try {
+    util.compileDocs(user, repo).then(() => {
+      util.serveDocs(app, user, repo);
+      res.send(`Successfully Compiled Docs! View them at /${user}/${repo}`);
+    });
+  } catch (error) {
+    res.send(
+      `Oops. We couldn't render these docs. Check your config if you have one and try again!`
+    );
+  }
```
Comment (ianjennings) [inline app.js:23]: Does `compileDocs` have an error handler? That would be better than `try / catch`

File: config.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "description": "Description",
+  "plugins": [],
+  "gaCode": ""
+}
```
Comment (ianjennings) [inline config.json:5]: The pull request description says this isn'' required right? Is this an example?

File: docsify.js [hunk 3]
```diff
@@ -1,5 +1,26 @@
...
+  }
+
+  // If no plugins, but config exists for description
+  return `<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"/><title>${config.repo}</title><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/><meta name="description" content="${config.description}"/> <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0"/> <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/docsify@4/lib/themes/vue.css"/> </head> <body> <div id="app"></div><script>window.$docsify={name: '${config.repo}', repo: '${config.user}/${config.repo}'}; </script> <script src="//cdn.jsdelivr.net/npm/docsify@4"></script> </body></html>`;
```
Comment (ianjennings) [inline docsify.js:23]: I feel like there's a way to do this without repeating all these HTML tags.

Comment (ianjennings) [reply docsify.js:23]: Can you turn `<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"/><title>${config.repo}</title><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"/><meta name="description" content="${config.description}"/> <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0"/> <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/docsify@4/lib/themes/vue.css"/> </head> <body> <div id="app"></div>` into a variable?

File: plugins.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    }
+    if (pluginList[i] == "google-analytics") {
+      pluginConfig.ga = true;
+    }
+    if (pluginList[i] == "emoji") {
+      scriptTags +=
+        "<script src='//cdn.jsdelivr.net/npm/docsify/lib/plugins/emoji.min.js'></script>";
+    }
+    if (pluginList[i] == "zoom-image") {
+      scriptTags +=
```
Comment (ianjennings) [inline plugins.js:26]: This is more like what I want

File: util.js [hunk 5]
```diff
@@ -40,8 +53,47 @@ function writeDocs(user, repo, readmeData) {
...
+
+    fs.writeFileSync(`./docs/${user}-${repo}/index.html`, htmlData, function (
+      err
+    ) {
+      if (err) throw err;
+      console.log("Generated HTML!");
+    });
+  } catch (error) {
+    writeDocsWithDefaultConfig(user, repo, readmeData);
+  }
```
Comment (ianjennings) [inline util.js:89]: `try/catch` is not a best practice. 
https://softwareengineering.stackexchange.com/questions/144326/try-catch-in-javascript-isnt-it-a-good-practice

File: util.js [hunk 6]
```diff
@@ -67,4 +119,34 @@ function serveDocs(app, user, repo) {
   app.use(`/${user}/${repo}`, express.static(`./docs/${user}-${repo}`));
 }
 
-module.exports = { getReadme, serveDocs };
+function isConfigDefault(config) {
+  if (config.name != "") return false;
```
Comment (ianjennings) [inline util.js:123]: Try `if(config.name.length)`

Comment (ianjennings) [reply util.js:123]: Or, `return config.name.length > 0`

File: config.json (hunk 2 above)
Comment (aemmadi) [reply config.json:5]: Yeah the `config.json` here is just the list of supported keys that can be used

File: util.js (hunk 5 above)
Comment (aemmadi) [reply util.js:89]: Hmm, interesting read. Will look more into error handling with callback functions. Future refactor!!

**Pull request merged**